src/Microsoft.Diagnostics.DebugServices.Implementation/Module.cs

```diff
@@ -38,17 +41,29 @@ public enum Flags : byte
 
         public readonly ServiceProvider ServiceProvider;
 
-        public Module()
+        public Module(ITarget target)
         {
             ServiceProvider = new ServiceProvider();
             ServiceProvider.AddServiceFactoryWithNoCaching<PEImage>(() => GetPEInfo());
-            ServiceProvider.AddServiceFactory<PEReader>(() => {
-                _onChangeEvent = ModuleService.SymbolService.OnChangeEvent.Register(() => ServiceProvider.RemoveService(typeof(PEReader)));
-                return ModuleService.GetPEReader(this);
+
+            ServiceProvider.AddServiceFactory<PEReader>(() => ModuleService.GetPEReader(this));
+            if (target.OperatingSystem == OSPlatform.Linux) {
+                ServiceProvider.AddServiceFactory<ELFFile>(() => ModuleService.GetELFFile(this));
+            }
+            if (target.OperatingSystem == OSPlatform.OSX) {
```

Comment:
We don't actually know what kind of module it is yet.  It can require a lot of "work" to figure it out (see ModuleService.GetPEInfo() for what we need to do for PEs).  The module/module service code lazily determines and caches all of this info.